src/components/AlteracaoDeCardapio/index.jsx

```diff
@@ -393,14 +536,42 @@ class AlteracaoCardapio extends Component {
     }
   }
 
+  exibeModalConfirmacao = values => {
+    const {
+      periodosQuePossuemLancheNaAlteracao,
+      ehAlteracaoComLancheRepetida
+    } = this.state;
+    if (
+      periodosQuePossuemLancheNaAlteracao.temRestricao &&
+      ehAlteracaoComLancheRepetida
+    ) {
+      values["eh_alteracao_com_lanche_repetida"] = true;
+      return this.showModalConfirm(values);
+    }
+    if (!ehAlteracaoComLancheRepetida) {
+      if (periodosQuePossuemLancheNaAlteracao.temRestricao) {
+        values["eh_alteracao_com_lanche_repetida"] = true;
+        return this.showModalConfirm(values);
+      } else {
+        values["eh_alteracao_com_lanche_repetida"] = false;
+        this.onSubmit(values);
+      }
+    } else {
+      values["eh_alteracao_com_lanche_repetida"] = false;
+      this.onSubmit(values);
+    }
```

Comment:
Dá pra simplificar assim:
if (!ehAlteracaoComLancheRepetida && periodosQuePossuemLancheNaAlteracao.temRestricao) {
    values["eh_alteracao_com_lanche_repetida"] = true;
    return this.showModalConfirm(values);
} else {
    values["eh_alteracao_com_lanche_repetida"] = false;
    this.onSubmit(values);
}